packages/stitch/tests/isolateFieldsFromSubschema.test.ts

```diff
@@ -91,13 +92,14 @@ describe('isolateComputedMergeSchemas', () => {
     });
   });
 
-  describe('from SDL deprecations', () => {
+  describe('from SDL directives', () => {
     const storefrontSchema = makeExecutableSchema({
       typeDefs: `
+        directive @requires(selectionSet: String, isolate: Boolean) on FIELD_DEFINITION
         type Product {
           id: ID!
-          shippingEstimate: Float! @deprecated(reason: "requires { price weight }")
-          deliveryService: DeliveryService! @deprecated(reason: "requires { weight }")
+          shippingEstimate: Float! @requires(selectionSet: "{ price weight }", isolate: true)
+          deliveryService: DeliveryService! @requires(selectionSet: "{ weight }", isolate: true)
```

Comment:
This becomes awkwardly similar but different from the federation spec: https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#requires.

I’d suggest supporting either “fields” or “selectionSet” properties; fields wouldn’t have wrapping curly braces. Then, the isolate property is really weird. The only reason it exists Is for an internal implementation detail which I think gets way more low-level than anyone will ever care to understand. I think just a requires attribute that isolates is majority use-case for MVP. If the soft-dependency feature were to be included in the SDL, I think it’d make more sense going through its own directive like “@sends(selectionSet:...)” that falls outside of the federation SDL spec and can be document as simply “like @requires, but doesn’t guarantee delivery of the field”. Just writing out that description there, I’m not convinced it has a strong use case.

Reply:
I like the fields or selectionSet option, one without, one with curly braces. We do need to support different syntax because we support a more complex possible set of required fields/subfields